src/user.ts

```diff
@@ -29,14 +29,29 @@ import { SecureBuffer } from './secureBuffer';
 import { SfdxError } from './sfdxError';
 import { sfdc } from './util/sfdc';
 
-const PASSWORD_LENGTH = 13;
 const LOWER = 'abcdefghijklmnopqrstuvwxyz';
 const UPPER = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
 const NUMBERS = '1234567890';
 const SYMBOLS = ['!', '@', '#', '$', '%', '^', '&', '*', '(', ')', '_', '[', ']', '|', '-'];
 
 const rand = (len: Many<string>): number => Math.floor(Math.random() * len.length);
 
+interface Complexity {
+  LOWER: number;
```

Comment:
you could also make these all optional unless true.  So `1` could be { LOWER: true } and 5 would be { LOWER:true, NUMBERS:true, SYMBOLS:true }